liquidprompt

```diff
@@ -1397,9 +1433,61 @@ _lp_kubernetes_context() {
 _lp_kubernetes_context_color() {
     _lp_kubernetes_context || return "$?"
 
-    lp_kubernetes_context_color="[${LP_MARK_KUBECONTEXT}${LP_COLOR_KUBECONTEXT}${lp_kubernetes_context}${lp_kubernetes_namespace+:}${lp_kubernetes_namespace-}${NO_COL}]"
+    lp_kubernetes_context_color="${LP_MARK_KUBECONTEXT}${LP_COLOR_KUBECONTEXT}${lp_kubernetes_context}${lp_kubernetes_namespace+:}${lp_kubernetes_namespace-}${NO_COL}"
 }
 
+_lp_cmake() {
+    (( LP_ENABLE_CMAKE )) || return 2
+    [[ -f CMakeCache.txt ]] || return 1
+
+    local cmake_compiler
+    IFS="=" read -r _ cmake_compiler <<<"$(grep COMPILER:FILEPATH CMakeCache.txt)"
+    lp_cmake_compiler="${cmake_compiler##*/}" # Only the part after the last slash.
+
+    local cmake_generator
+    IFS="=" read -r _ cmake_generator <<<"$(grep CMAKE_GENERATOR:INTERNAL CMakeCache.txt)"
+    # Shorten: Makefiles -> Make, and Visual Studio -> VS.
+    cmake_generator="${cmake_generator/%Makefiles/Make}"
+    cmake_generator="${cmake_generator/#Visual Studio/VS}"
+    # Remove all white spaces.
+    lp_cmake_generator="${cmake_generator// /}"
+
+    IFS="=" read -r _ lp_cmake_buildtype <<<"$(grep CMAKE_BUILD_TYPE:STRING CMakeCache.txt)"
+
+    [[ -n "${lp_cmake_compiler}${lp_cmake_generator}${lp_cmake_buildtype}" ]] || return 3
```

Comment:
This needs an answer before I can merge.

Reply:
It was just to have different error code for different issues, in case the calling function would need it. I did not think there was some semantics in the number. What would be the benefit of having two different issues leading to the same error code?

Reply:
I would argue that it is not two different issues, but two different paths to the same issue, being "no data available".

This is the standard for our data functions:

https://liquidprompt.readthedocs.io/en/stable/functions/data.html
> All data functions will return true (meaning return code 0) when they are both enabled and have data. They will return false (meaning return code 1) when they do not have data. Most will return 2 when they are disabled

I could see an argument that no `CMakeCache.txt` file is different from one existing but not containing the data we care about. But I think those semantics are too nuanced to try to indicate in a prompt. It is also not how the rest of our data functions work currently.